packages/fxa-payments-server/src/lib/stripe.ts

```diff
@@ -249,6 +252,12 @@ export async function handlePaymentIntent({
         paymentMethodId
       ) {
         apiDetachFailedPaymentMethod({ paymentMethodId });
+        // Delete incomplete subscription.
+        // If we do not do this, and a new subscription is added, with valid payment card,
+        // then both subscriptions will be paid, resulting in a duplicate charge.
+        // A new subscription will be created if the customer refreshes the page,
+        // instead of hitting the try again button.
+        apiDeleteSubscriptionId({ subscriptionId });
```

Comment:
Where is this `subscriptionId` from?  Can we delete on the back end without having the front end trigger it?

Reply:
Yup definitely. Doing it in the backend would probably be even better.